File: airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftConnectionHandler.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.redshift;

import java.sql.Connection;
import java.sql.SQLException;

public class RedshiftConnectionHandler {

  /**
   * For to close a connection. Aimed to be use in test only.
   *
   * @param connection The connection to close
   */
  public static void close(Connection connection) {
    try {
      connection.setAutoCommit(false);
      connection.commit();
      connection.close();
    } catch (SQLException e) {
      throw new RuntimeException(e);
    }
  }

}

File: airbyte-integrations/connectors/destination-redshift/src/test-integration/kotlin/io/airbyte/integrations/destination/redshift/RedshiftConnectionHandler.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.destination.redshift

import java.sql.Connection
import java.sql.SQLException

object RedshiftConnectionHandler {
    /**
     * For to close a connection. Aimed to be use in test only.
     *
     * @param connection The connection to close
     */
    fun close(connection: Connection) {
        try {
            connection.autoCommit = false
            connection.commit()
            connection.close()
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }
}
